tests/utils/loadable.test.tsx

```diff
@@ -256,6 +256,39 @@ it('loadable of a derived async atom with error does not trigger infinite loop (
   await findByText('Error: thrown in baseAtom')
 })
 
+it('does not repeatedly attempt to get the value of an unresolved promise atom wrapped in a loadable', async () => {
+  let resolveAsync: (value: number) => void
+  const baseAtom = atom(
+    new Promise<number>((resolve) => {
+      resolveAsync = resolve
+    })
+  )
+
+  let callsToGetBaseAtom = 0
+  const derivedAtom = atom((get) => {
+    callsToGetBaseAtom++
+    return get(baseAtom)
+  })
+
+  const { findByText } = render(
+    <StrictMode>
+      <Provider>
+        <LoadableComponent asyncAtom={derivedAtom} />
+      </Provider>
+    </StrictMode>
+  )
+
+  // depending on provider-less mode or versioned-write mode, there will be
+  // either 2 or 3 calls.
+  await findByText('Loading...')
+  expect(callsToGetBaseAtom).toBeLessThanOrEqual(3)
+
+  callsToGetBaseAtom = 0
+  resolveAsync!(5)
+  await findByText('Data: 5')
+  expect(callsToGetBaseAtom).toBe(1)
```

Comment:
Looks like we need "less than" here too?

Reply:
Added!